sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/SessionReceiverManager.cs

```diff
@@ -80,7 +82,11 @@ private async Task<bool> EnsureCanProcess(CancellationToken cancellationToken)
             {
                 await WaitSemaphore(cancellationToken).ConfigureAwait(false);
                 releaseSemaphore = true;
-                if (_threadCount >= _maxCallsPerSession)
+                if (_threadCount >= _maxCallsPerSession ||
+                    // If a receive call timed out for this session, avoid adding more threads
```

Comment:
I'd suggest hoisting the comment outside of the `if` statement.   I found it confusing and had to read a couple of times before it clicked for me that it was still part of the preceding conditional.